csharp-basic-topics/DetectKeyPressInCSharp/Tests/ConsoleAppTests.cs

```diff
@@ -0,0 +1,50 @@
+using DetectKeyPressInCSharp;
+using Moq;
+using NUnit.Framework;
+
+namespace Tests
+{ 
+    [TestFixture]
+    public class ConsoleAppTests
+    {
+        private Mock<IConsoleService> _mockConsoleService;
+        private ConsoleApp _app;
+
+        [SetUp]
+        public void Setup()
+        {
+            _mockConsoleService = new Mock<IConsoleService>();
+            _app = new ConsoleApp(_mockConsoleService.Object);
+        }
+
+        [Test]
+        public void KeyInputs_ReadKeyUseIsCalled_ShouldPrintCorrectMessages()
+        {
+            _mockConsoleService.SetupSequence(m => m.ReadKey(false))
+                .Returns(new ConsoleKeyInfo('A', ConsoleKey.A, false, false, false))
+                .Returns(new ConsoleKeyInfo('B', ConsoleKey.B, false, false, false));
+
+            _app.ReadKeyUse();
+
+            _mockConsoleService.Verify(m => m.WriteLine("Press any key to continue..."), Times.Exactly(2));
+            _mockConsoleService.Verify(m => m.WriteLine("\nYou pressed: A"), Times.Once());
+            _mockConsoleService.Verify(m => m.WriteLine("\nYou pressed: B"), Times.Once());
+            _mockConsoleService.Verify(m => m.WriteLine("Process stopped"), Times.Once());
+        }
+
+        [Test]
+        public void ApplicationIsRunning_WhenXIsPressed_ApplicationShouldStop()
```

Comment:
@mohitkumar342 ,
we also need to make some adjustment here.
`WhenXKeyIsPressed_ThenApplicationShouldStop`

Reply:
The test name is updated. Thanks